lib/scala/syntax/definition/src/main/scala/org/enso/syntax/text/ast/meta/Builtin.scala

```diff
@@ -259,20 +259,27 @@ object Builtin {
         case _ => internalError
       }
     }
+
     val (qualifiedImport, itemsImport, qualifiedExport, itemsExport) = {
-      val qualNamePat = Pattern.SepList(Pattern.Cons(), Opr("."))
-      val rename      = Var("as") :: Pattern.Cons()
+      val qualNamePat = (Pattern.fromAST(Var("project")) | ((Pattern
+        .Var() | Pattern.Cons()) :: Opr(".") :: Pattern.Cons())) :: (Opr(
+        "."
+      ) :: Pattern.SepList(Pattern.Cons(), Opr("."))).opt
+      val rename = Var("as") :: Pattern.Cons()
 
       def extractRename(
         matched: Pattern.Match
-      ): (List1[AST.Ident.Cons], Option[AST.Ident.Cons]) = {
+      ): (List1[AST.Ident], Option[AST.Ident.Cons]) = {
         val (nameMatch, renameMatch) =
           matched.asInstanceOf[Match.Seq[Shifted[AST]]].elem
-        val name: List1[AST.Ident.Cons] =
+        val name: List1[AST.Ident] =
           List1(
             nameMatch.toStream
               .map(_.wrapped)
-              .flatMap(AST.Ident.Cons.any.unapply)
+              .collect {
+                case AST.Ident.Cons.any(n) => n
+                case AST.Ident.Var.any(n)  => n
+              }
```

Comment:
Do you mean to implicitly ignore anything that somehow slips through to here? `.collect` will discard things that don't match.

Reply:
I think given how the `qualNamePat` is defined, nothing else could show up here, right?
But then if it is ever changed, indeed an error could be introduced that will go unnoticed here, so I think I'd +1 that it would be good to somehow handle the unexpected (currently, but not necessarily forever, impossible) case.

Reply:
It also contains the `Opr('.')` segments, this is a way of filtering them.